org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/xsd/contentmodel/CMXSDElementDeclaration.java

```diff
@@ -309,6 +311,32 @@ private void collectElementsDeclaration(XSComplexTypeDefinition typeDefinition,
 		}
 	}
 
+	public boolean isOptional(String childElementName) {
+		if (elementOptionality == null){
+			this.elementOptionality = new HashMap<String, Boolean>();
+			XSTypeDefinition typeDefinition = elementDeclaration.getTypeDefinition();
+			switch (typeDefinition.getTypeCategory()) {
+				case XSTypeDefinition.SIMPLE_TYPE:
+					break;
+				case XSTypeDefinition.COMPLEX_TYPE:
+					XSParticle particle = ((XSComplexTypeDefinition) typeDefinition).getParticle();
+					if (particle != null) {
+						XSObjectList objectList = ((XSModelGroup) particle.getTerm()).getParticles();
+						for (int i = 0; i < objectList.getLength(); i++) {
+							XSParticle xp = (XSParticle) objectList.item(i);
+							XSTerm t = xp.getTerm();
+							if (t instanceof XSElementDeclaration) {
+								if (xp != null) {
+									elementOptionality.put(t.getName(), xp.getMinOccurs() == 0);
+								}
+							}
+						}
+					}
+			}
+		}
+		return elementOptionality.get(childElementName);

```

Comment:
@datho7561 , @JessicaJHee

```
[Error - 9:44:43 a.m.] Aug. 17, 2022 09:44:43 org.eclipse.lemminx.services.XMLCodeActions doCodeActions()
Message: Error while processing code actions for the participant 'org.eclipse.le
mminx.extensions.contentmodel.participants.ContentModelCodeActionParticipant'.
java.lang.NullPointerException: Cannot invoke "java.lang.Boolean.booleanValue()" because the return value of "java.util.Map.get(Object)" is null
    at org.eclipse.lemminx.extensions.xsd.contentmodel.CMXSDElementDeclaration.isOptional(CMXSDElementDeclaration.java:338)
    at org.eclipse.lemminx.extensions.contentmodel.utils.XMLGenerator.isGenerateChild(XMLGenerator.java:405)
    at org.eclipse.lemminx.extensions.contentmodel.utils.XMLGenerator.generate(XMLGenerator.java:150)
    at org.eclipse.lemminx.extensions.contentmodel.utils.XMLGenerator.generate(XMLGenerator.java:124)
    at org.eclipse.lemminx.extensions.contentmodel.utils.XMLGenerator.generate(XMLGenerator.java:107)
    at org.eclipse.lemminx.extensions.contentmodel.participants.codeactions.cvc_complex_type_2_4_bCodeAction.doCodeAction(cvc_complex_type_2_4_bCodeAction.java:99)
    at org.eclipse.lemminx.extensions.contentmodel.participants.ContentModelCodeActionParticipant.doCodeAction(ContentModelCodeActionParticipant.java:51)
    at org.eclipse.lemminx.services.XMLCodeActions.doCodeActions(XMLCodeActions.java:60)
    at org.eclipse.lemminx.services.XMLLanguageService.doCodeActions(XMLLanguageService.java:247)
    at org.eclipse.lemminx.XMLTextDocumentService.lambda$codeAction$23(XMLTextDocumentService.java:464)
    at org.eclipse.lemminx.commons.ModelTextDocuments.lambda$computeModelAsync$0(ModelTextDocuments.java:118)
    at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646)
    at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:483)
    at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
    at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
    at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
    at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
    at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
```